crates/torii/graphql/src/object/erc/token_balance.rs

```diff
@@ -103,7 +103,11 @@ async fn fetch_token_balances(
          JOIN tokens t ON b.token_id = t.id
          JOIN contracts c ON t.contract_address = c.contract_address"
     );
-    let mut conditions = vec!["b.account_address = ?".to_string()];
+    let mut conditions = vec![
+        "(b.account_address = ?)".to_string(),
+        "t.metadata IS NOT NULL".to_string(),
+        "t.metadata != ''".to_string(),
```

Comment:
Do we want to ignore the tokens without metadata? We may return them and the front will see that they are missing, or do we have an other reason?

Reply:
there are few cases like nft got burned in which case the metadata would no longer be present, but we still need them in db since the token id is foreign key in other tables.

and also in some cases contracts don't follow the spec properly or the json metadata is broken, etc... even in those cases we still want indexing to happen normally and we would just ignore those tokens

Reply:
But only metadata is ignored, or the whole token? The whole token, right?

Reply:
yeah whole token, because without metadata there is no attributes, images, etc... so i dont think UI can do much with the token anyways